pkg/cmd/pulumi/pulumi.go

```diff
@@ -186,55 +267,94 @@ func NewPulumiCmd() *cobra.Command {
 	cmd.PersistentFlags().StringVar(
 		&color, "color", "auto", "Colorize output. Choices are: always, never, raw, auto")
 
-	// Common commands:
-	//     - Getting Started Commands:
-	cmd.AddCommand(newNewCmd())
-	//     - Deploy Commands:
-	cmd.AddCommand(newUpCmd())
-	cmd.AddCommand(newPreviewCmd())
-	cmd.AddCommand(newDestroyCmd())
-	cmd.AddCommand(newWatchCmd())
-	//     - Stack Management Commands:
-	cmd.AddCommand(newStackCmd())
-	cmd.AddCommand(newConfigCmd())
-	//     - Service Commands:
-	cmd.AddCommand(newLoginCmd())
-	cmd.AddCommand(newLogoutCmd())
-	cmd.AddCommand(newWhoAmICmd())
-	//     - Policy Management Commands:
-	cmd.AddCommand(newPolicyCmd())
-	//     - Advanced Commands:
-	cmd.AddCommand(newCancelCmd())
-	cmd.AddCommand(newImportCmd())
-	cmd.AddCommand(newRefreshCmd())
-	cmd.AddCommand(newStateCmd())
-	//     - Other Commands:
-	cmd.AddCommand(newLogsCmd())
-	cmd.AddCommand(newPluginCmd())
-	cmd.AddCommand(newVersionCmd())
-	cmd.AddCommand(newConsoleCmd())
-	cmd.AddCommand(newAboutCmd())
-	cmd.AddCommand(newSchemaCmd())
-	cmd.AddCommand(newOrgCmd())
-
-	// Less common, and thus hidden, commands:
-	cmd.AddCommand(newGenCompletionCmd(cmd))
-	cmd.AddCommand(newGenMarkdownCmd(cmd))
-
-	// We have a set of commands that are still experimental and that are hidden unless PULUMI_EXPERIMENTAL is set
-	// to true.
-	cmd.AddCommand(newQueryCmd())
-	cmd.AddCommand(newConvertCmd())
-
-	// We have a set of options that are useful for developers of pulumi that are hidden unless PULUMI_DEBUG_COMMANDS is
-	// set to true.
-	cmd.PersistentFlags().StringVar(&tracingHeaderFlag, "tracing-header", "",
-		"Include the tracing header with the given contents.")
+	setCommandGroups(cmd, []commandGroup{
+
+		// Common commands:
+		{
+			Name: "Deployment Commands",
+			Commands: []*cobra.Command{
+				newNewCmd(),
+				newConfigCmd(),
```

Comment:
Are `new` and `config` deployment commands? 🤔 (I know we debated the categories a bunch, so I just pile in)

Reply:
This would be confusing to me, as this would imply that somebody who is not getting started, shouldn't look into this section? I don't think we can call `pulumi up` a "getting started" command. It's a command for everyone.

Reply:
If you had to pick a header for grouping

`pulumi new`
`pulumi config`

what would you have in mind?